rs/xnet/hyper/src/lib.rs

```diff
@@ -355,9 +113,11 @@ impl TlsConnector {
 }
 
 impl Service<Uri> for TlsConnector {
-    type Response = TlsConnection;
+    type Future = Pin<
+        Box<dyn Future<Output = Result<MaybeHttpsStream<TokioIo<TcpStream>>, BoxError>> + Send>,
+    >;
+    type Response = MaybeHttpsStream<TokioIo<TcpStream>>;
```

Comment:
Can't we use the hyper client with hyper_rustls here?

Reply:
Yup I could not find a way because of the new client config required for each connection